### PR TITLE
version: support ignored shrinkwrap and package-locks

### PR DESCRIPTION
**Pull request by zkat**
Fixes: https://github.com/npm/npm/issues/17168

### CODE REVIEW

**Comment by iarna** [comment]
I wish the "I search and replaced var to const" part was a separate commit. It's v hard to see what your changes are as it is.

**Comment by zkat** [comment]
@iarna `+exports.exec = BB.promisify(execGit)` is the only actual change in that file.

I can pull that out and just do it separately, tho 👍

File: lib/version.js [hunk 1]
```diff
@@ -278,26 +278,33 @@ function checkGit (localData, cb) {
...
-    cb
-  )
+    }
+  }).nodeify(cb)
+}
+
+function addLocalFile (file, options, ignoreFailure) {
+  const p = git.exec(['add', path.join(npm.localPrefix, file)], options)
+  return ignoreFailure
+  ? p.catch(() => {})
```
Comment (iarna) [inline lib/version.js:306]: Ah, this is the new bit! 👍

**Review by iarna — APPROVED** [review]

**Comment by patrykkopycinski** [comment]
regression in `5.3.0`

**Pull request merged**